src/pyhs3/model.py

```
@@ -256,6 +273,11 @@ def log_prob(self) -> TensorVar:
                 if aux_name in self.distributions:
                     terms.append(pt.log(self.distributions[aux_name]))
 
+        # HFDC constraint terms: collected once per unique nuisance parameter
+        # across all channels during graph construction.
+        for constraint_expr in self._hfdc_constraints:
+            terms.append(pt.log(constraint_expr))
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)